docs/src/index.md

```diff
@@ -1,14 +1,185 @@
+
+
+There are several ways you can create these colors. An easy approach is to define the type through a "template" object:
+
+```jldoctest demo
+julia> ctemplate = ColorMixture{Float32}((RGB(0,1,0), RGB(1,0,0)))
+(0.0₁, 0.0₂)
 ```
 
-```@autodocs
-Modules = [FluorophoreColors]
+`ctemplate` is an all-zeros `ColorMixture` object, but can be used to construct arbitrary `c` with specified intensities:
+
+```jldoctest demo
+julia> typeof(ctemplate)
+ColorMixture{Float32, 2, (RGB{N0f8}(0.0,1.0,0.0), RGB{N0f8}(1.0,0.0,0.0))}
+
+julia> c = ctemplate(0.2, 0.4)
+(0.2₁, 0.4₂)
+
+julia> Tuple(c)
+(0.2f0, 0.4f0)
+```
+
+You can also create them with a single call `ColorMixture(rgbs, intensities)`:
+
+```jldoctest demo
+julia> c = ColorMixture{Float32}((RGB(0,1,0), RGB(1,0,0)), (0.2, 0.4))
+(0.2₁, 0.4₂)
+```
+
+or even by explicit type construction:
+
+```jldoctest demo
+julia> ColorMixture{Float32, 2, (RGB{N0f8}(0.0,1.0,0.0), RGB{N0f8}(1.0,0.0,0.0))}(0.2, 0.4)
+(0.2₁, 0.4₂)
 ```
+
+!!! tip
+    All but the last form require [constant propagation](https://en.wikipedia.org/wiki/Constant_folding) for inferrability.
+    Julia 1.7 and higher can use "aggressive" constant propagation to solve inference problems that may reduce performance on Julia 1.6.
+
+### Importing external data
+
+When objects are not created by code but instead loaded from an external source such as a file, you have several avenues for creating arrays of multichannel color objects. There are two particularly common cases:
+
+1. If the imported data are an array `A` of size `(nc, m, n)`, where `nc` is the number of color channels (i.e., color is the fastest dimension), then use `reinterpret(reshape, C, A)` where `C` is the color type you want to use (e.g., `MultiChannelColor{T,nc}` or `ColorMixture{T,nc,rgbs}`). For instance, Landsat 8 data might look something like this:
+
+   ```julia
+   A = rand(0x0000:0x0fff, 11, 100, 100);
+   img = reinterpret(reshape, MultiChannelColor{N4f12,11}, A);
+   ```
+
+2. If the imported data have the color channel last, or use separate arrays for each channel, use the [StructArrays package](https://github.com/JuliaArrays/StructArrays.jl). For example:
+
+   ```julia
+   A = rand(0x0000:0x0fff, 100, 100, 11);
+   img = StructArray{MultiChannelColor{N4f12,11}}(A; dims=3)
+   ```
+
+## Additional features
+
+### Fluorophores
+
+This package also exports a lookup table for common [fluorophores](https://en.wikipedia.org/wiki/Fluorophore). If desired, these can be used as the `rgbⱼ` values for `ColorMixture` channels. For example:
+
+```jldoctest demo
+julia> channels = (fluorophore_rgb["EGFP"], fluorophore_rgb["tdTomato"])
+(RGB{N0f8}(0.0,0.925,0.365), RGB{N0f8}(1.0,0.859,0.0))
+
+julia> ctemplate = ColorMixture{N0f16}(channels)
+(0.0N0f16₁, 0.0N0f16₂)
+```
+
+If you'll be hard-coding the name of the fluorophore, consider using a slightly different syntax:
+
+```jldoctest demo
+julia> channels = (fluorophore_rgb"EGFP", fluorophore_rgb"tdTomato")
```

Comment:
Will `fluorophore"EGFP"` be easier to use? I'm not sure. -- just that I have never seen underscores used in string macros.

Reply:
No non-RGB fluorophore currently. I guess I would think that `fluorophore"EGFP"` might return an entire object that is a complete characterization of the fluorophore, e.g., full spectra, cross section, quantum yield, and so on ([this list](https://en.wikipedia.org/wiki/Fluorophore#Fluorescent_proteins) only scratches the surface). In contrast, `fluorophore_rgb` makes it clear that what you're getting is an RGB value that represents the fluorophore.